### PR TITLE
MULE-13970: not recalculating value for null result

### PR DESCRIPTION
**Pull request by mbuchwald**

### CODE REVIEW

File: src/main/java/org/mule/mvel2/optimizers/impl/refl/nodes/NullSafe.java [hunk 1]
```diff
@@ -22,45 +23,26 @@ public NullSafe(char[] expr, int start, int offset, ParserContext pCtx) {
...
+    // This is intended to avoid an infinite recursion.
+    // A null safe bean property should return null in
+    // order to end the evaluation process when the
+    // there is no key found.
+    if (innerAccessor instanceof NullSafe) {
+      return null;
+    }
 
+    Object value = innerAccessor.getValue(ctx, elCtx, variableFactory);
+    if (nextNode!=null && value !=null) {
```
Comment (dfeist) [inline src/main/java/org/mule/mvel2/optimizers/impl/refl/nodes/NullSafe.java]: couple of missing spaces.

File: src/test/java/org/mule/mvel2/tests/core/PropertyAccessTests.java [hunk 2]
```diff
@@ -604,4 +608,135 @@ public void testAccessPropertyInsideForWithOptionals() {
...
+  public void testAccessWithOptionalMapOfOptionalMap() {
+    Map<String, Object> a = new HashMap<String, Object>();
+    Map<String, Map<String, Object>> m = singletonMap("a", a);
+
+    Map<String, Map<String, String>> nestMap = singletonMap("foo", singletonMap("bar", "foobar"));
+
+    String ex = "a.?inner.?foo.?bar";
+
+    Serializable s = MVEL.compileExpression(ex);
+    //assertNull(MVEL.executeExpression(s, m));  // If I decomment this, the test passes
```
Comment (dfeist) [inline src/test/java/org/mule/mvel2/tests/core/PropertyAccessTests.java]: remove or uncomment

File: src/test/java/org/mule/mvel2/tests/core/PropertyAccessTests.java [hunk 3]
```diff
@@ -604,4 +608,135 @@ public void testAccessPropertyInsideForWithOptionals() {
...
+    String ex = "a.?inner[0].foo.bar";
+
+    Serializable s = MVEL.compileExpression(ex);
+    assertNull(MVEL.executeExpression(s, m));
+    a.put("inner", nestMap);
+    assertEquals("foobar", MVEL.executeExpression(s, m));
+  }
+
+  @Ignore
+  public void testAccessOptionalListWithOptionalMapOfMap() {
```
Comment (dfeist) [inline src/test/java/org/mule/mvel2/tests/core/PropertyAccessTests.java]: Create issues? Or are you unsure this case is even valid?

File: src/test/java/org/mule/mvel2/tests/core/PropertyAccessTests.java [hunk 4]
```diff
@@ -604,4 +608,135 @@ public void testAccessPropertyInsideForWithOptionals() {
...
+    Serializable s = MVEL.compileExpression(ex);
+    assertNull(MVEL.executeExpression(s, m)); // If I comment this, the test passes
+    a.put("inner", singletonList(Collections.emptyMap()));
+    assertNull(MVEL.executeExpression(s, m));
+    a.put("inner", nestList);
+    assertEquals("foobar", MVEL.executeExpression(s, m));
+  }
+
+  @Ignore
+  public void testAccessWithOptionalMapOfOptionalMap() {
```
Comment (dfeist) [inline src/test/java/org/mule/mvel2/tests/core/PropertyAccessTests.java]: Create issues? Or are you unsure this case is even valid?

File: src/test/java/org/mule/mvel2/tests/core/PropertyAccessTests.java (hunk 3 above)
Comment (mbuchwald) [reply src/test/java/org/mule/mvel2/tests/core/PropertyAccessTests.java]: I created the issue, I will link it to test, but I'm not completely sure the scenario is valid (the issue itself mention this)

**Pull request merged**